ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm

```diff
@@ -243,4 +281,53 @@ - (void)testShouldHandleVnodeOpEvent {
     XCTAssertEqual(kauthResult, KAUTH_RESULT_DEFER);
 }
 
+/*
+KEXT_STATIC int HandleVnodeOperation(
+    kauth_cred_t    credential,
+    void*           idata,
+    kauth_action_t  action,
+    uintptr_t       arg0,
+    uintptr_t       arg1,
+    uintptr_t       arg2,
+    uintptr_t       arg3);
+*/
+- (void)testHandleVnodeOpEvent {
+    // Setup
+    kern_return_t initResult = VirtualizationRoots_Init();
+    XCTAssertEqual(initResult, KERN_SUCCESS);
+
+    // Parameters
+    const char* repoPath = "/Users/test/code/Repo";
+    const char* filePath = "/Users/test/code/Repo/file";
+    vfs_context_t _Nonnull context = vfs_context_create(NULL);
+    PerfTracer perfTracer;
+    PrjFSProviderUserClient dummyClient;
+    pid_t dummyClientPid=100;
+
+    // Create Vnode Tree
+    shared_ptr<mount> testMount = mount::Create();
+    shared_ptr<vnode> repoRootVnode = testMount->CreateVnodeTree(repoPath, VDIR);
+    shared_ptr<vnode> testFileVnode = testMount->CreateVnodeTree(filePath);
+
+    // Register provider for the repository path (Simulate a mount)
+    VirtualizationRootResult result = VirtualizationRoot_RegisterProviderForPath(&dummyClient, dummyClientPid, repoPath);
+    XCTAssertEqual(result.error, 0);
+    vnode_put(s_virtualizationRoots[result.root].rootVNode);
```

Comment:
@pmj I saw the same lines in your unit test.  How will the system clean this up for real scenarios?

Reply:
This is normally done in `ActiveProvider_Disconnect()` which last I checked wasn't quite ready to be called from unit tests yet. You could give it a go if you like (and the situation may have since improved).